lib/resolve-shims.js

```diff
@@ -177,7 +177,7 @@ var resolve = module.exports = function resolveShims (file, messages, cb) {
         , shim               :  shimsByPath[file]
         , exposeGlobals      :  cached.exposeGlobals
         , browser            :  pack.browser
-        , 'browserify-shim'  :  pack['browserify-shim']
+        , 'browserify-shim'  :  Object.assign({},pack['browserify-shim']||{},shimConfig)
```

Comment:
This looks rather complex, what does this do? Also let's not use ES6 only syntax to ensure this keeps working with older node versions please.